### ISSUE TITLE
{} style array initializers aren't recognized

### ISSUE THREAD

**Issue by dannypurcell** [Darkyenus/glsl4idea#55]
While working with a 430 core comp shader I noticed a couple of incorrect syntax checks.

* In the following line the space between `image2d` and `framebuffer` is marked with `Expected '{'`
`layout(binding = 0, rgba32f) uniform image2D framebuffer;`

* In the following snippet multiple errors are marked but the root cause is likely that { } style initializers are not recognized correctly. 
```
const box boxes[] = {
  /* The ground */
  {vec3(-5.0, -0.1, -5.0), vec3(5.0, 0.0, 5.0)},
  /* Box in the middle */
  {vec3(-0.5, 0.0, -0.5), vec3(0.5, 1.0, 0.5)}
};
```

Samples from a recent [LWJGL Ray trace tutorial](https://github.com/LWJGL/lwjgl3-wiki/wiki/2.6.1.-Ray-tracing-with-OpenGL-Compute-Shaders-%28Part-I%29).
Syntax validates with the [Reference Compiler](https://www.khronos.org/opengles/sdk/tools/Reference-Compiler/)

**Comment by dannypurcell** [comment]
Thank you for taking on this project! 
I took a shot at it a while back but I could not spare the time to make meaningful improvements here while also trying to learn OpenGL in the first place.

If you have Gratipay or some other kind of donation system setup I'd like to send you something for taking this on.

**Comment by Darkyenus** [comment]
Thanks for the feedback!
I appreciate your willingness to donate, but I am not sure if I should set it up, since we have multiple core contributors (I didn't even start this project), so it wouldn't be just right to set it up without any kind of split. Maybe others can say what they think, but I am not sure about it. Thanks nevertheless.

Your first reported issue should be now fixed by #59 .
